### PR TITLE
Step2

### PR DESCRIPTION
**Pull request by andriimikhaylovskiy**

### CODE REVIEW

File: src/main/java/mate/academy/spring/boot/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
```
Comment (olekskov) [inline src/main/java/mate/academy/spring/boot/controller/BookController.java]: Better to move api to application.properties as global prefix in order not to mention it in every controller
```suggestion
@RequestMapping("/books")
```

File: src/main/java/mate/academy/spring/boot/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        this.title = title;
+    }
+
+    public String getAuthor() {
+        return author;
+    }
+
+    public void setAuthor(String author) {
+        this.author = author;
+    }
```
Comment (olekskov) [inline src/main/java/mate/academy/spring/boot/dto/CreateBookRequestDto.java]: Why do we need Lombok in dependencies if we don't use it?

File: src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public BookDto getBookById(Long id) {
+        return null;
+    }
+
+    @Override
+    public BookDto createBook(CreateBookRequestDto bookDto) {
+        return null;
+    }
```
Comment (olekskov) [inline src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java]: Please do not include methods without implementation

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/boot/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/books")
+public class BookController {
+
```
Comment (okuzan) [inline src/main/java/mate/academy/spring/boot/controller/BookController.java]:
```suggestion
```

File: src/main/java/mate/academy/spring/boot/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (okuzan) [inline src/main/java/mate/academy/spring/boot/dto/CreateBookRequestDto.java]: use @ Data annotation

File: src/main/java/mate/academy/spring/boot/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+@Entity
```
Comment (okuzan) [inline src/main/java/mate/academy/spring/boot/model/Book.java]: it's not advisable to use @Data with entities, they can have side effects, if you don't implement equals and hashcode by yourself

File: src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import mate.academy.spring.boot.dto.BookDto;
+import mate.academy.spring.boot.dto.CreateBookRequestDto;
+import mate.academy.spring.boot.model.Book;
+import mate.academy.spring.boot.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+
```
Comment (okuzan) [inline src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/boot/model/Book.java [hunk 7]
```diff
@@ -6,9 +6,15 @@
...
-@Data
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
```
Comment (okuzan) [inline src/main/java/mate/academy/spring/boot/model/Book.java]: the problem with data is that it includes this annotation, so you didn't resolve the problem
if you don't need equals at this point, just remove it

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**